Client/Client.Qt/Widgets/ChatHistory.cpp

```diff
@@ -20,7 +20,7 @@ ChatHistory::ChatHistory(QWidget* parent) : QWidget(parent), _messageList()
 void ChatHistory::addMessage(const QString& message, quint64 utc, const QString& user)
 {
     auto history = _scrollArea->widget();
-    auto msg     = new MessageWidget(history, message, utc, user);
+    auto msg     = new MessageWidget(history, message, _messageId++, _userId, utc, user);
```

Comment:
What for did you increment _messageId?

Reply:
It's not the first time when see that you increment and didn't explain why. Please, in such situations, leave comments.